src/main/kotlin/uk/gov/justice/hmpps/offenderevents/config/SchedulerConfiguration.kt

```diff
@@ -0,0 +1,22 @@
+package uk.gov.justice.hmpps.offenderevents.config
+
+import net.javacrumbs.shedlock.core.LockProvider
+import net.javacrumbs.shedlock.provider.jdbctemplate.JdbcTemplateLockProvider
+import net.javacrumbs.shedlock.spring.annotation.EnableSchedulerLock
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.scheduling.annotation.EnableScheduling
+import javax.sql.DataSource
+
+@Configuration
+@EnableScheduling
+@EnableSchedulerLock(
+  defaultLockAtLeastFor = "\${application.events.poll.lockAtLeastFor}",
```

Comment:
Apart from converting the file from Java to Kotlin - the only line changes was this was to pick up the lockAtLeastFor